pages/settings/index.js

```diff
@@ -87,6 +83,123 @@ export function SettingsHeader () {
+          <h5 className='card-title'>Delete Account</h5>
+          <p className='card-text'>
+            This will permanently delete your account. This action cannot be undone.
+          </p>
+
+          {!showConfirmation
+            ? (
+              <Button
+                variant='danger'
+                onClick={() => setShowConfirmation(true)}
+              >
+                Delete my account
+              </Button>
+              )
+            : (
+              <>
+                <Alert variant='danger'>
+                  <p><strong>Warning:</strong> Account deletion is permanent and cannot be reversed.</p>
+                  <p>Before proceeding, please ensure:</p>
+                  <ul>
+                    <li>You have withdrawn all your sats or checked the box to donate your balance to the rewards pool</li>
+                    <li>You understand that you will lose access to your account name</li>
+                    <li>You have considered that this action affects your entire account history</li>
+                  </ul>
+                </Alert>
+
+                <BootstrapForm.Check
+                  type='checkbox'
+                  id='delete-content'
+                  label='Also anonymize all my posts and comments (they will show as "[deleted]")'
+                  checked={deleteContent}
+                  onChange={(e) => setDeleteContent(e.target.checked)}
+                  className='mb-3'
+                />
+
+                <BootstrapForm.Check
+                  type='checkbox'
+                  id='donate-balance'
+                  label='Donate my remaining balance to the rewards pool (required if you have a balance)'
+                  checked={donateBalance}
+                  onChange={(e) => setDonateBalance(e.target.checked)}
+                  className='mb-3'
+                />
+
+                <BootstrapForm.Group className='mb-3'>
+                  <BootstrapForm.Label>Type "DELETE MY ACCOUNT" to confirm:</BootstrapForm.Label>
+                  <BootstrapForm.Control
+                    type='text'
+                    value={confirmation}
+                    onChange={(e) => setConfirmation(e.target.value)}
+                    placeholder='DELETE MY ACCOUNT'
+                  />
+                </BootstrapForm.Group>
+
+                <div className='d-flex justify-content-between'>
+                  <Button
+                    variant='secondary'
+                    onClick={() => setShowConfirmation(false)}
+                  >
+                    Cancel
+                  </Button>
+                  <Button
+                    variant='danger'
+                    disabled={confirmation !== 'DELETE MY ACCOUNT'}
+                    onClick={handleDelete}
+                  >
+                    Permanently delete my account
+                  </Button>
+                </div>
+              </>
+              )}
+        </div>
+      </div>
+    </>
+  )
```

Comment:
I think this should use our custom form with a schema for validation

(see other comment in api/resolvers/user.js)